### PR TITLE
Remove json fields for internal client config

### PR DESCRIPTION
**Pull request by yue9944882**
/assign @sttts @caesarxuchao 

/cc @adohe @dixudx 

/sig api-machinery

idk if its designed? 😥or its for a backward-compatibility?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by dixudx — COMMENTED** [review]
Normally we won't have json serialization/deserialization for such internal api types.

But this change is okay for me.

@sttts @deads2k @liggitt I think we should remove these json tags in this file.

**Comment by yue9944882** [comment]
@dixudx good catch! just had a sync with @sttts on slack, will raise another pull to remove the tags for internal types in other private schemas

File: staging/src/k8s.io/client-go/tools/clientcmd/api/types.go [hunk 1]
```diff
@@ -30,53 +30,53 @@ type Config struct {
 	// Legacy field from pkg/api/types.go TypeMeta.
 	// TODO(jlowdermilk): remove this after eliminating downstream dependencies.
 	// +optional
-	Kind string `json:"kind,omitempty"`
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/clientcmd/api/types.go]: I tried doing this a while ago and I'm pretty sure it broke some of the `kubectl config` commands

**Comment by liggitt** [comment]
The test failures demonstrate what this breaks

**Comment by liggitt** [comment]
/close

closing because of inactivity

**Pull request closed without merging**